bigquery_etl/glam/glean_metric_counts.py

```diff
@@ -0,0 +1,69 @@
+r"""Metric counting.
+
+```bash
+python3 -m bigquery_etl.glam.glean_metric_counts
+
+diff \
+    <(cat sql/telemetry_derived/clients_histogram_probe_counts_v1/query.sql) \
+    <(python3 -m bigquery_etl.glam.glean_metric_counts)
+```
+"""
+from itertools import combinations
+from typing import List
+
+from jinja2 import Environment, PackageLoader
+
+from bigquery_etl.format_sql.formatter import reformat
+
+
+def render_query(
+    header: str,
+    source_table: str,
+    attributes: List[str],
+    aggregate_attributes: str,
+    aggregate_grouping: str,
+    **kwargs,
+) -> str:
+    """Render the main query."""
+    env = Environment(loader=PackageLoader("bigquery_etl", "glam/templates"))
+    sql = env.get_template("metric_counts_v1.sql")
+
+    # include attributes included and excluded from grouping set
+    attribute_combinations = []
+    for subset_size in reversed(range(len(attributes) + 1)):
+        for grouping in combinations(attributes, subset_size):
+            hidden = list(sorted(set(attributes) - set(grouping)))
+            attribute_combinations.append((grouping, hidden))
```

Comment:
This generates more lines than anticipated, what is the difference in logic from the handwritten one? @emtwo